app/src/main/java/com/example/tasky/core/data/remote/BaseRepository.kt

```diff
@@ -0,0 +1,53 @@
+package com.example.tasky.core.data.remote
+
+import com.example.tasky.core.data.Resource
+import com.squareup.moshi.Moshi
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import okhttp3.ResponseBody
+import retrofit2.HttpException
+import retrofit2.Response
+import java.io.IOException
+
+abstract class BaseRepository {
```

Comment:
Having a base class is way too overkill just to share a single function. It creates a lot of coupling and confusion (because nobody will be able to tell you what's inside `BaseRepository` without opening it).
A normal function in a plain Kotlin file will be totally enough 👍